examples/preprocessing/plot_xdawn_denoising.py

```diff
@@ -51,25 +51,33 @@
 raw.info['bads'] = ['MEG 2443']  # set bad channels
 picks = pick_types(raw.info, meg=True, eeg=False, stim=False, eog=False,
                    exclude='bads')
+
 # Epoching
 epochs = Epochs(raw, events, event_id, tmin, tmax, proj=False,
                 picks=picks, baseline=None, preload=True,
                 add_eeg_ref=False, verbose=False)
 
-# Plot image epoch before xdawn
-plot_image_epochs(epochs['vis_r'], picks=[230], vmin=-500, vmax=500)
+# Divide epochs into a train and a test set to avoid potential circular
+# analyses (a.k.a. double dipping) in later steps.
+train = range(len(epochs) / 2)
+test = range(len(epochs) / 2, len(epochs))
 
-# Estimates signal covariance
+# Estimates signal covariance on first half of the data
+half_time = epochs[train].events[-1, 0] / raw.info['sfreq']
+raw.crop(0, half_time)
 signal_cov = compute_raw_data_covariance(raw, picks=picks)
 
+# Plot image epoch before xdawn
+plot_image_epochs(epochs['vis_r'][test], picks=[230], vmin=-500, vmax=500)
+
 # Xdawn instance
 xd = Xdawn(n_components=2, signal_cov=signal_cov)
```

Comment:
a bit off topic, but i made another script using MNE : https://www.kaggle.com/alexandrebarachant/grasp-and-lift-eeg-detection/visual-evoked-potential-vep

Since the kaggle environment does not have the last version of MNE, i made a 20 line minimalistic implementation of XDAWN.
Since it's only a visualization example, i do not use CV :)